phpBB/includes/ucp/ucp_profile.php

```diff
@@ -170,27 +170,21 @@ function main($id, $mode)
 						{
 							$message = ($config['require_activation'] == USER_ACTIVATION_SELF) ? 'ACCOUNT_EMAIL_CHANGED' : 'ACCOUNT_EMAIL_CHANGED_ADMIN';
 
-							include_once($phpbb_root_path . 'includes/functions_messenger.' . $phpEx);
-
 							$server_url = generate_board_url();
 
 							$user_actkey = gen_rand_string(mt_rand(6, 10));
 
-							$messenger = new messenger(false);
-
+							$messenger = $phpbb_container->get('messenger.method_collection');
+							$email = $messenger->offsetGet('messenger.method.email');
```

Comment:
Not sure why you need the method collection then?

Reply:
To get the email service from it?

Reply:
Or just `$phpbb_container->get('messenger.method.email');`?

Reply:
Yes, the latter one should work just fine and IMHO makes more sense here